src/commons/nativeJS/EvalErrorLocator.ts

```diff
@@ -0,0 +1,86 @@
+import { SourceError } from 'js-slang/dist/types';
+
+enum BrowserType {
+  Chrome = 'Chrome',
+  FireFox = 'FireFox',
+  Unsupported = 'Unsupported'
+}
+
+interface EvalErrorLocator {
+  regex: RegExp;
+  browser: BrowserType;
+}
+// Chrome
+// Example: `TypeError: window.dsl is not a function at eval (eval at evalNativeJSProgram (NativeJSEval.ts:6:1), <anonymous>:1:8)`
+const ChromeEvalErrorLocator = {
+  regex: /at eval.+<anonymous>:(\d+):(\d+)/gm,
+  browser: BrowserType.Chrome
+};
+
+// Firefox
+// Example: @http://localhost:8000/static/js/main.chunk.js line 31925 > eval:1:1
+//            evalNativeJSProgram@http://localhost:8000/static/js/main.chunk.js:31925:16`
+const FireFoxEvalErrorLocator = {
+  regex: /eval:(\d+):(\d+)/gm,
+  browser: BrowserType.Chrome
+};
+
+const EVAL_ERROR_LOCATORS: EvalErrorLocator[] = [ChromeEvalErrorLocator, FireFoxEvalErrorLocator];
+
+function getBrowserType(): BrowserType {
+  const userAgent: string = navigator.userAgent.toLowerCase();
+  return userAgent.indexOf('chrome') > -1
+    ? BrowserType.Chrome
+    : userAgent.indexOf('firefox') > -1
+    ? BrowserType.FireFox
+    : BrowserType.Unsupported;
+}
+
+function extractErrorLocation(
+  errorStack: string,
+  lineOffset: number,
+  errorLocator: EvalErrorLocator
+): SourceError['location'] | undefined {
+  const evalErrors = Array.from(errorStack.matchAll(errorLocator.regex));
+  if (evalErrors.length) {
+    // we only want to get the initial eval error
+    // this is to account for nested `eval()` calls
+    const baseEvalError = evalErrors[evalErrors.length - 1];
+    const [lineNumStr, colNumStr] = baseEvalError.slice(1, 3);
+    return {
+      start: { line: parseInt(lineNumStr) - lineOffset, column: parseInt(colNumStr) },
+      end: { line: 0, column: 0 }
+    };
+  }
```

Comment:
refactored error locator to account for nested `eval()` calls

Example: 
```javascript
// this is within the editor running "full Javascript"
eval("const a = a;")
```

The above snippet would provide the following error stack:

```
Uncaught ReferenceError: Cannot access 'a' before initialization
    at eval (eval at <anonymous> (eval at <anonymous> (playground:1)), <anonymous>:1:11)
    at eval (eval at <anonymous> (playground:1), <anonymous>:1:1)
    at <anonymous>:1:1
```
This means that we will be getting the error line of the inner `eval()` (which I think would be a bit confusing?), therefore  I refactored it to extract all eval errors and get the outermost error. In the case where we would want to also show nested `eval()` we can also extend the function to support it.

> this is of course in the case where we allow user-submitted full JS programs to use `eval()` as well